DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js

```diff
@@ -85,12 +118,167 @@ function submitPresetData() {
+
+  // verify that user is logged in before submission.
+  if (userId == 0) {
+    alert("Login before submitting!");
+    return;
+  }
+  keyval_pairs.push(encodeURIComponent("userId") + "=" + encodeURIComponent(userId));
+
+  // verify that keywordCampaignId is set.
+  if (keywordCampaignId == null) {
+    alert("Select a keyword campaign before submitting!");
+    return;
+  }
+  keyval_pairs.push(encodeURIComponent("keywordCampaignId") + "=" + encodeURIComponent(keywordCampaignId));
+  
+  // default values for variables (not applicable to creation phase) sent to servlet
+  keyval_pairs.push(encodeURIComponent("DSACampaignId") + "=" + encodeURIComponent("0"));
+  keyval_pairs.push(encodeURIComponent("campaignStatus") + "=" + encodeURIComponent("pending"));
+  keyval_pairs.push(encodeURIComponent("clicks") + "=" + encodeURIComponent("0"));
+  keyval_pairs.push(encodeURIComponent("cost") + "=" + encodeURIComponent("0"));
+  keyval_pairs.push(encodeURIComponent("impressions") + "=" + encodeURIComponent("0"));
+  
+  // represents campaign location - built during runtime
+  let location = "";
+
+  var form = document.getElementById('campaign-form'); // get the comment form
+  for (var i = 0; i < form.elements.length; i++) {
+    // Form contains buttons that are irrelevant to input - need to filter out only input
+    if (form.elements[i].nodeName === "BUTTON") {
+      continue;
+    }
+
+    // stop submission process if parameter is empty.
+    if ((form.elements[i].value === null) || (form.elements[i].value === "")) {
+      console.log(form.elements[i].nodeName);
+      alert("Not all the settings are filled out!");
+      return;
+    }
+    
+    // build location string 'Region, Country' - country occurs in the form first.
+    if (form.elements[i].name.includes("region")) {
+      location = form.elements[i].value + "," + location;
+      keyval_pairs.push(encodeURIComponent("locations") + "=" + encodeURIComponent(location));
+    }
+    else if (form.elements[i].name.includes("country")) {
+      location = form.elements[i].value;
+    }
+    else {  
+      console.log(`value ${form.elements[i].name} ${form.elements[i].value}`);
+      keyval_pairs.push(encodeURIComponent(form.elements[i].name) + "=" + encodeURIComponent(form.elements[i].value));
+    }
+  }
+
+  // separate each entity in keyval_pairs with '&' for query string
+  var queryString = keyval_pairs.join("&");
+
+  xmlhttp.open("POST", '/DSA-campaigns', true);
+  xmlhttp.setRequestHeader("Content-type","application/x-www-form-urlencoded");
+  console.log(queryString);
+  xmlhttp.send(queryString);
+}
+
+/**
+ * processes each onchange event associated with the keyword-campaigns
+ * select element. If element value is 0 (no selected element) then form 
+ * does not show. If not 0 (selected element), then form shows.
+ */
+function keywordSelection() {
+  var select_element_value = document.getElementById("keyword-campaigns").value;
+  if (select_element_value == 0) {
```

Comment:
nit: select_element_value should be selectElementValue to fit with the style

Reply:
Fixed